sql/planbuilder/builder.go

```diff
@@ -25,12 +25,13 @@ type scope struct {
 	ast    ast.SQLNode
 	node   sql.Node
 
-	cols      []scopeColumn
-	extraCols []scopeColumn
-	tables    map[string]tableId
-	ctes      map[string]*scope
-	groupBy   *groupBy
-	exprs     map[string]columnId
+	cols        []scopeColumn
+	extraCols   []scopeColumn
+	redirectCol map[string]scopeColumn
```

Comment:
How is this different from aliasing?

Reply:
This might miss some edge cases, but aliases are new expressions that contend for name resolution and can be GetField referenced. NATURAL JOIN right columns need to be redirected to pre-existing name definitions, and not contend with the attribute from the left-table.